tests/HeaderFactoryTest.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Aidantwoods\SecureHeaders\Tests;
+
+use Aidantwoods\SecureHeaders\HeaderFactory;
+use PHPUnit_Framework_TestCase;
+
+class HeaderFactoryTest extends PHPUnit_Framework_TestCase
+{
+    public function testCSPHeaders()
+    {
+        $CSPHeader = HeaderFactory::build(
+            'Content-SECURITY-Policy',
+            "default-src 'none'"
+        );
+
+        $CSPROHeader = HeaderFactory::build(
+            'ConTeNt-Security-Policy-Report-Only',
+            "default-src 'none'"
+        );
+
+        $XCSPHeader = HeaderFactory::build(
+            'x-Content-Security-POLICY',
+            "default-src 'none'"
+        );
+
+        $XCSPROHeader = HeaderFactory::build(
+            'X-content-security-policy-report-only',
+            "default-src 'none'"
+        );
+
+        $Headers = [$CSPHeader, $CSPROHeader, $XCSPHeader, $XCSPROHeader];
+
+        foreach ($Headers as $Header)
+        {
+            $this->assertSame(
+                'Aidantwoods\SecureHeaders\Headers\CSPHeader',
```

Comment:
Thanks for taking a look!

I had a look at that, I figured it worked like PHP's `instanceof` though? (which isn't what I want – interested in the exact class).

Would work okay for `CSPHeader` right now, but not for `RegularHeader` (as `CSPHeader` extends `RegularHeader` – so any `CSPHeader` is an `instanceof` `RegularHeader`).